Sources/MapboxMaps/Annotations/ViewAnnotationManager.swift

```diff
@@ -219,9 +219,12 @@ public final class ViewAnnotationManager {
         }
         let currentFeatureId = try? mapboxMap.options(forViewAnnotationWithId: id).associatedFeatureId
         try mapboxMap.updateViewAnnotation(withId: id, options: options)
-        let isHidden = !(options.visible ?? true)
-        expectedHiddenByView[view] = isHidden
-        viewsById[id]?.isHidden = isHidden
+
+        if let visible = options.visible, visible == false {
```

Comment:
It is the same, isn't it?
```suggestion
        if visible == false {
```

Reply:
And if it's not, can we add some tests to catch the corner case, please? This Optional<Bool> logic is really tricky 😅

Reply:
Same thing indeed, not sure why I went with more verbose syntax